utils/src/main/scala/za/co/absa/enceladus/utils/fs/FileSystemVersionUtils.scala

```diff
@@ -144,6 +145,38 @@ class FileSystemVersionUtils(conf: Configuration) {
     fs.getContentSummary(hdfsPath).getLength
   }
 
+  /**
+    * Returns directory size in bytes, skipping hidden files and directories (starting from '_' or '.').
+    *
+    * @param path A path to a directory or a file.
+    * @return Directory size in bytes
+    */
+  def getDirectorySizeNoHidden(path: String): Long = {
+    def getDirSizeHelper(f: Path): Long = {
+      val status = fs.getFileStatus(f)
+      if (status.isFile) {
+        status.getLen
+      } else {
```

Comment:
This is only applicable for the outer function call. Otherwise we end up checking if a dir (which we already know from line 164) is a file with each recursive call.